src/Components/Users/users.jsx

```diff
@@ -1,44 +1,101 @@
 import React from 'react';
+import * as axios from 'axios';
 
-import { I18N } from '../../constants';
-import usersArray from '../../usersArray';
+import { I18N, API } from '../../constants';
 
 import style from './users.module.css';
 
-const users = (props) => {
-  if (props.users.length === 0) {
-    props.setUsers(usersArray);
+import defaultImg from '../Image/person.jpg';
+
+class Users extends React.Component {
+  state = {
+    pages: [],
+  };
+  componentDidMount() {
+    const {
+      setUsers, setTotalUsersCount, currentPage, pageSize,
+    } = this.props;
+
+    axios.get(`${API.BASE_URL}${API.USER_RESOURCE}${API.PAGE}${currentPage}${API.COUNT}${pageSize}`)
+      .then(({ data }) => {
+        const { items, totalCount } = data;
+        setUsers(items);
+        setTotalUsersCount(totalCount);
+        this.a();
+      });
+  }
+  a = () => {
+    const pages = [];
+    for (let i = 1; i <= this.pagesCount(); i++) {
+      pages.push(i);
+    }
+    this.setState({ pages })
   }
-  const toggleButton = (followed, id) => (followed ? <button type="button" onClick={() => { props.unfollowUser(id); }}>{I18N.ENG.UNFOLLOW}</button>
-    : <button type="button" onClick={() => { props.followUser(id); }}>{I18N.ENG.FOLLOW}</button>);
 
-  const usersItem = ({
-    id, photoUrl, followed, fullname, status, location,
-  }) => (
-    <div key={id} className={style.item}>
+  onPageSelected = (pageNumber) => {
+    const { setUsers, setCurrentPage, pageSize, } = this.props;
+
+    setCurrentPage(pageNumber);
+    axios.get(`${API.BASE_URL}${API.USER_RESOURCE}${API.PAGE}${pageNumber}${API.COUNT}${pageSize}`)
+      .then(({data}) => {
+        const { items } = data;
+        setUsers(items);
+      });
+  };
+  createSpans = () => {
+    const { currentPage } = this.props;
+
+    return (
       <div>
-        <img src={photoUrl} alt="" className={style.img} />
-        {toggleButton(followed, id)}
+        {this.state.pages.map((p,index) => (
```

Comment:
Consider putting this JSX into separate function "createSpan".
Then in "createSpans" function you'll have mapping for `this.state.pages` using "createSpan" as a map executor.

Reply:
i've done it